src/utils/binning-utils.js

```diff
@@ -0,0 +1,86 @@
+import NUMBERS from '../constants/numbers';
+
+const DefaultResolutionLevel = 6;
+const MaxResolutionLevel = 8;
+
+const splitJSONDataPages = (dp) => {
+  const bins = [];
+  const matrix = dp[0].qMatrix;
+  let i;
+
+  bins.push(matrix[0][0]);
+
+  // Hack for snapshot
+  if (dp[0].reformatted) {
+    for (i = 1; i < matrix.length; i++) {
+      try {
+        bins.push(matrix[i][0]);
+      } catch (err) {
+        // console.log( err );
+      }
+    }
+  } else {
+    for (i = 1; i < matrix.length; i++) {
+      try {
+        matrix[i][0].qText = JSON.parse(matrix[i][0].qText);
+        bins.push(matrix[i][0]);
+      } catch (err) {
+        // console.log( err );
+      }
+    }
+    // eslint-disable-next-line no-param-reassign
+    dp[0].reformatted = true;
+  }
+  return [bins];
+};
+
+const getBinnedData = (left, top, width, height, zoomLevel, layoutService, model) => {
```

Comment:
I suggest we move the two functions `getBinnedData` and `updateBinnedData` out from this generic utils file (should avoid creating those...). And move it either into the dataFetcher or a new file which we call `binned-data-fetcher`.